File: server/java-api/myproject/src/main/java/com/javaapi/app/user/usecase/Session/SessionFactory.java
package com.javaapi.app.user.usecase.Session;

import org.springframework.stereotype.Service;

import com.javaapi.app.user.core.dto.SessionDTO;
import com.javaapi.app.user.framework.session.SessionStore;


@Service
public class SessionFactory{

    private final SessionStore sessionStore;
    public SessionFactory(SessionStore sessionStore) {
        this.sessionStore = sessionStore;
    }

    public void createUserSession(SessionDTO sessionDTO,String session) {

        sessionStore.saveSession(session, sessionDTO.getUserId(), sessionDTO.getUsername(), sessionDTO.getEmail());
    }
}

File: server/go-ws/internal/app/usecase/session/SessionFactory.go
package session

import (
	"go-ws/internal/app/core/domain/model/vo/session"
	dto "go-ws/internal/app/core/dto/session"
)

type SessionFactory struct{}

func NewSessionFactory() *SessionFactory {
	return &SessionFactory{}
}

func (sf *SessionFactory) CreateSession(sessiondto *dto.SessionDTO) *dto.SessionDTO {

	userId := sessiondto.GetUserID()
	username := sessiondto.GetUsername()
	email := sessiondto.GetEmail()

	validUserId := session.NewUserID(userId)
	validUsername := session.NewUsername(username)
	validEmail := session.NewEmail(email)

	sessionDTO := dto.NewSessionDTO(
		validUserId.GetUserID(),
		validUsername.GetUsername(),
		validEmail.GetEmail(),
	)

	return sessionDTO
}
